File: src/main/java/com/fatec/destino/dto/compra/ViagemDetalhadaDTO.java
package com.fatec.destino.dto.compra;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record ViagemDetalhadaDTO(
        UUID id,
        String nomePacote,
        String descricao,
        BigDecimal valor,
        String statusCompra,
        @JsonFormat(pattern = "dd-MM-yyyy")
        LocalDate dataPartida,
        @JsonFormat(pattern = "dd-MM-yyyy")
        LocalDate dataRetorno,
        @JsonFormat(pattern = "dd-MM-yyyy")
        LocalDate dataCompra,
        String numeroReserva,
        String imagemPrincipal,
        List<String> galeria,
        List<String> inclusoes,
        String nomeHotel,
        String tipoTransporte
) {}

File: src/main/kotlin/com/fatec/destino/dto/compra/ViagemDetalhadaDTO.kt
package com.fatec.destino.dto.compra

import com.fatec.destino.dto.pacote.hotel.HotelRegistroDTO
import com.fatec.destino.model.pacote.pacoteFoto.foto.Foto
import com.fatec.destino.util.model.transporte.Meio
import java.math.BigDecimal
import java.time.LocalDate
import java.util.*

data class ViagemDetalhadaDTO(
    val id: UUID?,
    val nomePacote: String?,
    val descricao: String?,
    val valor: BigDecimal?,
    val statusCompra: String?,
    val dataPartida: LocalDate?,
    val dataRetorno: LocalDate?,
    val dataCompra: Date,
    val numeroReserva: String?,
    val imagemPrincipal: String?,
    val galeria: MutableSet<Foto?>?,
    val inclusoes: MutableList<String?>?,
    val nomeHotel: HotelRegistroDTO,
    val tipoTransporte: Meio?
)
